From 7c342e38adc501175310a81cdedf6d9bae0f7c36 Mon Sep 17 00:00:00 2001
From: Yann Sionneau <yann.sionneau@vates.tech>
Date: Thu, 30 Oct 2025 17:16:16 +0100
Subject: [PATCH] Add debug messages in xc_dom_gnttab_init in libxenguest

---
 tools/libs/guest/xg_dom_boot.c | 7 +++++++
 1 file changed, 7 insertions(+)

diff --git a/tools/libs/guest/xg_dom_boot.c b/tools/libs/guest/xg_dom_boot.c
index 5c7e12221d..0495c3895b 100644
--- a/tools/libs/guest/xg_dom_boot.c
+++ b/tools/libs/guest/xg_dom_boot.c
@@ -325,6 +325,8 @@ static int compat_gnttab_hvm_seed(xc_interface *xch, uint32_t domid,
         .domid = domid,
     };
 
+    xc_dom_printf(xch, "%s() d%d", __func__, domid);
+
     rc = xc_core_arch_get_scratch_gpfn(xch, domid, &scratch_gfn);
     if ( rc < 0 )
     {
@@ -385,12 +387,15 @@ int xc_dom_gnttab_seed(xc_interface *xch, uint32_t guest_domid,
     xenforeignmemory_resource_handle *fres;
     void *addr = NULL;
 
+    xc_dom_printf(xch, "%s() d%d", __func__, guest_domid);
+
     fres = xenforeignmemory_map_resource(
         fmem, guest_domid, XENMEM_resource_grant_table,
         XENMEM_resource_grant_table_id_shared, 0, 1, &addr,
         PROT_READ | PROT_WRITE, 0);
     if ( !fres )
     {
+        xc_dom_printf(xch, "%s() d%d xenforeignmemory_map_resource() failed", __func__, guest_domid);
         if ( errno == EOPNOTSUPP )
             return is_hvm ?
                 compat_gnttab_hvm_seed(xch, guest_domid,
@@ -422,6 +427,8 @@ int xc_dom_gnttab_init(struct xc_dom_image *dom)
     xen_pfn_t console_gfn = xc_dom_p2m(dom, dom->console_pfn);
     xen_pfn_t xenstore_gfn = xc_dom_p2m(dom, dom->xenstore_pfn);
 
+    xc_dom_printf(dom->xch, "%s() d%d", __func__, dom->guest_domid);
+
     return xc_dom_gnttab_seed(dom->xch, dom->guest_domid, is_hvm,
                               console_gfn, xenstore_gfn,
                               dom->console_domid, dom->xenstore_domid);
-- 
2.43.0

